src/vs/workbench/services/history/browser/historyService.ts

```diff
@@ -198,8 +198,10 @@ export class HistoryService extends Disposable implements IHistoryService {
 		this.handleActiveEditorChangeInNavigationStacks(group, editorPane);
 	}
 
-	private handleActiveEditorSelectionChangeEvent(group: IEditorGroup, editorPane: IEditorPaneWithSelection, event: IEditorPaneSelectionChangeEvent): void {
-		this.handleActiveEditorSelectionChangeInNavigationStacks(group, editorPane, event);
+	private handleActiveEditorSelectionChangeEvent(group: IEditorGroup, editorPane: IEditorPaneWithSelection, event: IEditorPaneSelectionChangeEvent[]): void {
```

Comment:
I wonder how much we gain from an array of events, because when I look at this, we check for the event reason and then at one point probably go into the pane and ask for the selection. This is now at the risk of simply looping multiple times asking for the same selection. So instead of accumulating this event, why not debounce it truly and then only ask once?

Reply:
That's what it would have done before though, I didn't want to change behavior if I could avoid it as it could cause regressions. If you can debounce properly that seems fine to me, I think you would need to merge the events though. You would also probably still want to accumulate the events but merge them after so the diffing/merging doesn't impact input latency.

Tbh this feels like a micro-optimization that won't actually yield anything in practice, my main interest is getting as much as possible out of the space between a keydown event and the animation frame finishing. Remember the screenshot is not a single keystroke but many keystrokes, looping over and handling each of the events will probably take millisecond(s) for a single call which is a very small price when deferred to after the animation frame.